File: go/136_Single_Number/136_Single_Number.go
func singleNumber(nums []int) int {
    m := make(map[int]int)

    for _, n := range nums {
        _, exists := m[n]
        if exists {
            delete(m, n)
        } else {
            m[n] += 1
        }
    }
    for k := range m {
        return k
    }
    return 0
}

File: java/136_Single_Number/136_Single_Number.java
class Solution {
    public int singleNumber(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        
        for (int n : nums) {
            if (map.containsKey(n)) {
                map.remove(n);
            } else {
                map.merge(n, 1, Integer::sum);
            }
        }
        for (int key : map.keySet()) {
            return key;
        }
        return 0;
    }
}
